tiny_cnn/io/tensorflow/proto_parser.h

```diff
@@ -0,0 +1,256 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <map>
+#include "tensorflow/core/framework/graph.pb.h"
+using namespace std;
+using namespace tensorflow;
+
+template<class in_value>
+void all2str(string & result, const in_value& t)
+{
+    ostringstream oss;
+    oss<<t;
+    result=oss.str();
+}
+
+static float char_to_float(const char* b) {
+  float f;
+  memcpy(&f, b, sizeof(float));
+  return f;
+}
+
+static float char_to_int32(const char* b) {
+  int32_t i;
+  memcpy(&i, b, sizeof(int32_t));
+  return i;
+}
+
+void summarize_attr_value(const string& attr_string, const AttrValue& attr_value) {
```

Comment:
@Wangyida 
We will violate the one definition rule (ODR) if we declare function without inline/static. see:
http://en.cppreference.com/w/cpp/language/definition